pkg/workspace/tuning/preset-tuning.go

```diff
@@ -182,35 +182,34 @@ func CreatePresetTuning(ctx context.Context, workspaceObj *kaitov1beta1.Workspac
 
 	// TODO: make containers only mount the volumes they need
 
-	// Add shared volume for training output
-	trainingOutputVolume, trainingOutputVolumeMount, outputDir := SetupTrainingOutputVolume(ctx, configVolume)
+	// Add volume for training output
+	outputVolume := workspaceObj.Tuning.Output.Volume
+	trainingOutputVolume, trainingOutputVolumeMount, outputDir := SetupTrainingOutputVolume(ctx, configVolume, outputVolume)
 	volumes = append(volumes, trainingOutputVolume)
 	volumeMounts = append(volumeMounts, trainingOutputVolumeMount)
 
 	initContainer, dataSourceVolumes, dataSourceVolumeMounts := prepareDataSource(ctx, workspaceObj)
 	volumes = append(volumes, dataSourceVolumes...)
 	volumeMounts = append(volumeMounts, dataSourceVolumeMounts...)
-	if initContainer.Name != "" {
+	if initContainer != nil && initContainer.Name != "" {
 		initContainers = append(initContainers, *initContainer)
 	}
 
-	sidecarContainer, imagePushSecret, dataDestVolume, dataDestVolumeMount := prepareDataDestination(ctx, workspaceObj, outputDir)
-	volumes = append(volumes, dataDestVolume)
-	volumeMounts = append(volumeMounts, dataDestVolumeMount)
+	sidecarContainer, imagePushSecret, imagePushSecretVolume, imagePushSecretVolumeMount := prepareDataDestination(ctx, workspaceObj, outputDir)
+	volumes = append(volumes, imagePushSecretVolume...)
+	volumeMounts = append(volumeMounts, imagePushSecretVolumeMount...)
 	if sidecarContainer != nil {
-		sidecarContainers = append(sidecarContainers, *sidecarContainer)
+		pauseContainer := corev1.Container{
+			Name:            "pause",
+			Image:           "registry.k8s.io/pause:latest",
+			ImagePullPolicy: corev1.PullAlways,
+		}
+		sidecarContainers = append(sidecarContainers, pauseContainer, *sidecarContainer)
 	}
 	if imagePushSecret != nil {
 		imagePullSecrets = append(imagePullSecrets, *imagePushSecret)
 	}
 
-	pauseContainer := corev1.Container{
-		Name:            "pause",
-		Image:           "registry.k8s.io/pause:latest",
-		ImagePullPolicy: corev1.PullAlways,
-	}
-	sidecarContainers = append(sidecarContainers, pauseContainer)
```

Comment:
These lines of code are hard to understand. Together with the 4 outputs params of `prepareDataDestination`.

What about make them two explicit cases? for example

```
if workspaceObj.Tuning.Output.Volume != nil {
volumes = xx
volumeMount = xx
sidecarContainers  = xx
else if workspaceObj.Tuning.Output.Image != "" {
volumes = xx
volumeMount = xx
sidecarContainers  = xx
}
```

Reply:
The code change here is not very large, I tends to keep the original logic as it is, maybe we could make the refactor in another standalone pr.

Reply:
The code change here is not very large, I tends to keep the original logic as it is, maybe we could make the refactor in another standalone pr.